### PR TITLE
util: support load Colon-separated KUBECONFIG

### LINKED ISSUE #437: error when KUBECONFIG contains more than one file
my `KUBECONFIG` contains a `:` delimited list of config files for `kubectl` to load. I think this is causing issues.

```bash
$ docker buildx create    \
  --driver kubernetes     \  
  --driver-opt replicas=3 \
  --use
```
```bash
$ docker buildx ls         
NAME/NODE           DRIVER/ENDPOINT             STATUS PLATFORMS
gallant_mahavira *  kubernetes                         
  gallant_mahavira0 unix:///var/run/docker.sock cannot determine Kubernetes namespace, specify manually: stat /home/$USER/.kube/config:/home/$USER/.kube/microk8s.config: no such file or directory
default             docker                              
  default           default              
```

I have yet to get a working k8s version running so not sure if I am just missing something obvious.

### PR DESCRIPTION
**Pull request by morlay**
fix #437 #760

### CODE REVIEW

File: commands/util.go [hunk 1]
```diff
@@ -246,14 +246,15 @@ func driversForNodeGroup(ctx context.Context, dockerCli command.Cli, ng *store.N
...
+			_ = os.Setenv(clientcmd.RecommendedConfigPathEnvVar, kubeconfig)
+		}
+		rules := clientcmd.NewDefaultClientConfigLoadingRules()
+		apiConfig, err := rules.Load()
+		if err != nil {
+			return nil, err
 		}
+		return clientcmd.NewDefaultClientConfig(*apiConfig, &clientcmd.ConfigOverrides{}), nil
 	}
 	return kubernetes.ConfigFromContext(endpointName, s)
```
Comment (morlay) [inline commands/util.go:259]: Should we still need this?

`github.com/docker/cli/cli/context/kubernetes` not provide a way to load `Colon-separated KUBECONFIG`
now changes could support all KUBECONFIG, and will not execute this line any more

**Comment by Dentrax** [comment]
```bash
$ echo $KUBECONFIG
:/Users/furkan.turkal/.kube/config:/Users/furkan.turkal/.kube/foo:/Users/furkan.turkal/.kube/bar:/Users/furkan.turkal/.kube/baz
```

This returns me the following error for [this commit](https://github.com/docker/buildx/pull/761/commits/c83812144ca594e2a5dff0437cdf1a3c2fe9ed92):

```
Error: could not read kubeconfig: stat : no such file or directory
```

**Comment by morlay** [comment]
@Dentrax 

this pr should fix it.
the new kubeconfig load follow kubectl did

**Comment by morlay** [comment]
why your kubeconfig value startswith `:`

**Review by AkihiroSuda — APPROVED** [review]

**Comment by Dentrax** [comment]
> why your kubeconfig value startswith `:`

🤷‍♂️I removed `:` prefix and still getting same error. 🤔

**Comment by morlay** [comment]
```
$ docker buildx create --name=kubeconfig --driver=kubernetes --driver-opt=namespace=gitlab
kubeconfig
$ docker buildx inspect kubeconfig                                                        
Name:   kubeconfig
Driver: kubernetes

Nodes:
Name:      kubeconfig0
Endpoint:  kubernetes:///kubeconfig?deployment=&kubeconfig=%2FUsers%2Fmorlay%2F.kube%2Fconfig--hw-dev.yaml%3A%2FUsers%2Fmorlay%2F.kube%2Fconfig--hw-infra.yaml%3A%2FUsers%2Fmorlay%2F.kube%2Fconfig--hw-sg.yaml%3A%2FUsers%2Fmorlay%2F.kube%2Fconfig--qc-hk.yaml
Status:    inactive
Platforms: 
$ docker buildx inspect kubeconfig --bootstrap
...
```

Works well for me.

**Comment by tonistiigi** [comment]
Should we remove empty files?

**Comment by morlay** [comment]
> Should we remove empty files?

`kubectl` didn't, so i don't think we sould add this file exists check.

For me,  my `KUBECONFIG` set like this to make sure files ok
```
export KUBECONFIG=$(echo $(ls ~/.kube/config--*.yaml) | sed 's/ /:/g')
```

**Comment by Dentrax** [comment]
@morlay Just applied your pull patch in my own fork and it seems working now but:

```bash
$ kubectl config current-context
kind-cluster-api-test
```

```bash
$ go run cmd/buildx/main.go create --name=kubeconfig --driver=kubernetes --driver-opt=namespace=gitlab
kubeconfig
```

```bash
$ go run cmd/buildx/main.go inspect kubeconfig
Name:   kubeconfig
Driver: kubernetes

Nodes:
Name:      kubeconfig0-787cbbd5d9-d2d9w
Endpoint:
Status:    running
Platforms: linux/amd64, linux/386
```

But getting the following issue and not sure if it's related to this, any ideas?

```
$ cat ~/.docker/buildx/instances/kubeconfig
{"Name":"kubeconfig","Driver":"kubernetes","Nodes":[{"Name":"kubeconfig0","Endpoint":"kubernetes:///kubeconfig?deployment=\u0026kubeconfig=%3A%2FUsers%2Ffurkan.turkal%2F.kube%2Fconfig%3A%2FUsers%2Ffurkan.turkal%2F.kube%2Ffoo%3A%2FUsers%2Ffurkan.turkal%2F.kube%2Fbar%3A%2FUsers%2Ffurkan.turkal%2F.kube%2Fbaz%3A%2FUsers%2Ffurkan.turkal%2Fauth%2Fk8s-auto%2FS_PLATFORM_P1_6PLATFORM_MOON","Platforms":null,"Flags":null,"ConfigFile":"","DriverOpts":{"namespace":"gitlab"}}],"Dynamic":false}

$ kubectl config get-contexts
CURRENT   NAME                                            CLUSTER                        AUTHINFO           NAMESPACE
*         kubernetes-admin@s-platform-p1-6platform-moon   s-platform-p1-6platform-moon   kubernetes-admin

$ kubectl cluster-info
Kubernetes control plane is running at ...
Metrics-server is running at ...
...

$ kubectl get pods -n gitlab
NAME                           READY   STATUS    RESTARTS   AGE
kubeconfig0-787cbbd5d9-d2d9w   1/1     Running   0          27m

$ ./buildx build -t test:simple-golang-app -f Dockerfile .
[+] Building 0.0s (0/0)
Error: no valid drivers found: cannot determine Kubernetes namespace, specify manually: invalid configuration: [context was not found for specified context: kubernetes-admin@s-platform-p1-6platform-moon, cluster has no server defined]
```

**Comment by morlay** [comment]
@Dentrax 

Why your `kubectl config current-context` not match `kubectl config get-contexts` ?

**Comment by tonistiigi** [comment]
@AkihiroSuda still lgty?

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**